platform/openide.filesystems/src/org/openide/filesystems/FileUtil.java

```diff
@@ -2337,6 +2341,29 @@ private static boolean isArchiveFileImpl(final URL url, final boolean strict) {
         }
         return archiveRootProviderCache;
     }
+    
+    /**
+     * Hardcoded offending filename characters. Windows are more picky than Unixes; relying
+     * on just the current host OS will limit file portability when shared through VCS.
+     */
+    private static final Pattern ILLEGAL_FILENAME_CHARACTERS = Pattern.compile("[\\/:\"*?<>|]");
+    
+    /**
+     * Determines whether the string forms a valid filename (without a path component).
+     * @param fileName candidate string
+     * @return true, if the string can be used to name a file.
+     */
```

Comment:
This is an API change and it comes with no versioning.

Reply:
Well. I was about task if, in your opinion, is this utility worth adding at all. But I've found ~9 occurrences of invalid filename characters (they differ, naturally ;)) 
In addition, I have found this interesting post - https://stackoverflow.com/questions/1976007/what-characters-are-forbidden-in-windows-and-linux-directory-names ... really, you just can't make `PRN` folder under Windows :-D so I guess we could eventually fit more checks even in the `FileUtil.findFreeFileName` ...
So I'll keep the utility + [NETBEANS-5720](https://issues.apache.org/jira/browse/NETBEANS-5720), [NETBEANS-5721](https://issues.apache.org/jira/browse/NETBEANS-5721) in addition.

Reply:
Addressed in #b623dd4b129c

Reply:
Also that would be better to be added in a different PR. This one is getting a bit bloated...

Reply:
@lkishalmi the initial impl of the utility is a separate commit here. The retrofit **will** be a separate PR (as well as the possible extension of the checks). I wonder :)) what happens when I place `CON` files to git on Linux and clone it to Windows ... ;)